### ISSUE TITLE
BoxPlot macro bug / requirement for 'ordinal' ?

### ISSUE THREAD

**Issue by michaelaye** [vega/vega-lite#5259]
I'm wondering if this is a bug or a `feature`, that one has to use `type='ordinal` to make the boxplot work properly?

Using this spec:
```json
{
  "data": {"url": "https://vega.github.io/vega-datasets/data/population.json"},
  "mark": {"type": "boxplot", "extent": "min-max"},
  "encoding": {
    "x": {"type": "quantitative", "field": "age"},
    "y": {"type": "quantitative", "field": "people"}
  }
}
```
creates an odd looking plot where bins seem to be merged:

![visualization](https://user-images.githubusercontent.com/69774/62419015-3e392d80-b634-11e9-8136-d75fe6ba6763.png)

while using this spec (only changed x-type to `ordinal`):

```json
{
  "data": {"url": "https://vega.github.io/vega-datasets/data/population.json"},
  "mark": {"type": "boxplot", "extent": "min-max"},
  "encoding": {
    "x": {"type": "ordinal", "field": "age"},
    "y": {"type": "quantitative", "field": "people"}
  }
}
```

creates this neat looking plot:

![visualization (1)](https://user-images.githubusercontent.com/69774/62419202-0b913400-b638-11e9-8cb7-451f30f963ea.png)

My use case is that I need boxplots to plot boxes over irregularly sampled points in time, instead of the usual ordinal categorical versions that plotting engines usually only offer for boxplots.

**Comment by domoritz** [comment]
Interesting use case. The spec works as expected since with two quantitative axes, we don't know what the orientation of the boxplot is. Do you want to rewrite your bug report as a feature request?

**Comment by michaelaye** [comment]
This has been closed via #5263

**Comment by domoritz** [comment]
Great. Fixed in https://github.com/vega/vega-lite/pull/5274.
